xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -49,28 +49,63 @@ using namespace PVR;
 
 CGUIDialogPVRTimerSettings::CGUIDialogPVRTimerSettings(void)
   : CGUIDialogSettingsManualBase(WINDOW_DIALOG_PVR_TIMER_SETTING, "DialogPVRTimerSettings.xml"),
+    m_bIsRadio(false),
+    m_bIsRepeating(false),
+    m_bSupportsFolders(false),
+    m_iWeekdays(0),
+    m_iPriority(0),
+    m_iLifetime(0),
     m_tmp_iFirstDay(0),
     m_tmp_day(11),
     m_bTimerActive(false),
-    m_selectedChannelEntry(0),
-    m_timerItem(NULL)
+    m_selectedChannelEntry(0)
 {
   m_loadType = LOAD_EVERY_TIME;
 }
 
 void CGUIDialogPVRTimerSettings::SetTimer(CFileItem *item)
 {
-  m_timerItem         = item;
+  m_InfoTag = item->GetPVRTimerInfoTag();
 
-  m_timerItem->GetPVRTimerInfoTag()->StartAsLocalTime().GetAsSystemTime(m_timerStartTime);
-  m_timerItem->GetPVRTimerInfoTag()->EndAsLocalTime().GetAsSystemTime(m_timerEndTime);
-  m_timerStartTimeStr   = m_timerItem->GetPVRTimerInfoTag()->StartAsLocalTime().GetAsLocalizedTime("", false);
-  m_timerEndTimeStr     = m_timerItem->GetPVRTimerInfoTag()->EndAsLocalTime().GetAsLocalizedTime("", false);
+  // Copy over data from tag. We must not change the tag until Save()
+  // because it is not a copy, but the (one and only) original.
+
+  m_bIsRadio         = m_InfoTag->m_bIsRadio;
+  m_bIsRepeating     = m_InfoTag->m_bIsRepeating;
+  m_iWeekdays        = m_InfoTag->m_iWeekdays;
+  m_iPriority        = m_InfoTag->m_iPriority;
+  m_iLifetime        = m_InfoTag->m_iLifetime;
+  m_strTitle         = m_InfoTag->m_strTitle;
+  m_strDirectory     = m_InfoTag->m_strDirectory;
+
+  m_FirstDay         = m_InfoTag->FirstDayAsLocalTime();
+  m_StartTime        = m_InfoTag->StartAsLocalTime();
+  m_EndTime          = m_InfoTag->EndAsLocalTime();
+
+  m_bTimerActive     = m_InfoTag->IsActive();
+  m_bSupportsFolders = m_InfoTag->SupportsFolders();
+
+  m_Channel          = m_InfoTag->ChannelTag();
+
+  m_StartTime.GetAsSystemTime(m_timerStartTime);
+  m_EndTime.GetAsSystemTime(m_timerEndTime);
+  m_timerStartTimeStr = m_StartTime.GetAsLocalizedTime("", false);
+  m_timerEndTimeStr   = m_EndTime.GetAsLocalizedTime("", false);
+
+  m_selectedChannelEntry = 0;
+  m_channelEntries.clear();
 
   m_tmp_iFirstDay     = 0;
   m_tmp_day           = 11;
 }
 
+void CGUIDialogPVRTimerSettings::SetChannelFromSelectedEntry(int iSelectedEntry, bool bRadio)
+{
+  const std::map<std::pair<bool, int>, int>::const_iterator itc(m_channelEntries.find(std::make_pair(bRadio, iSelectedEntry)));
```

Comment:
Not actually related to this PR, but makes perfect sense (I love C++11). Will change all places accordingly. Thx for the headsup.